### PR TITLE
fix(avatars): image backgrounds and grouped icon and image sizes

### PR DESCRIPTION
**Pull request by SiTaggart**
### Avatar Image background 

Ensures a transparent background for avatars that display an image that is transparent. Also adds a shadow border to visually match other avatar variants

### Pass size to group children

Sizes for icon and image avatars in a group was only ever passed to the parent, so the design asset never scaled to the size. This ensure the icon or image gets relevant size dimenstions as well.

## Checklist

- [ ] Work in a "Draft" branch whilst you are getting feedback to reduce Chromatic costs. Once you are ready for approval, convert the PR to "Ready to review"
- [ ] Ensure all `required` checks have passed
- [ ] Icon changes must have a product design review
- [ ] Website content changes must have a product design review
- [ ] At least two engineers have reviewed any code changes
- [ ] At least one engineering lead has reviewed any core package changes or repository infrastructure
- [ ] Website visual regression testing is run by adding `🕵🏻‍♀️ Run website visual regression` label

### CODE REVIEW

File: packages/paste-core/components/avatar/src/Avatar.tsx [hunk 1]
```diff
@@ -123,11 +123,13 @@ export const Avatar = React.forwardRef<HTMLDivElement, AvatarProps>(
         flexShrink={0}
         ref={ref}
         size={groupSize || size}
-        {...(src ? undefined : shadowVariants[color])}
+        {...(src ? {boxShadow: 'shadowBorderWeaker'} : shadowVariants[color])}
         {...colorVariants[color]}
         {...variants[groupVariant || variant]}
+        // when its an image the background should be transparent, for when you are using transparent images
+        {...(src && {backgroundColor: 'transparent'})}
```
Comment (TheSisb) [inline packages/paste-core/components/avatar/src/Avatar.tsx:130]: praise: good fix! 

question: can this be merged into line 126 or is the order in which these styles applied brittle?

**Review by nkrantz — APPROVED** [review]

File: packages/paste-core/components/avatar/src/Avatar.tsx (hunk 1 above)
Comment (SiTaggart) [reply packages/paste-core/components/avatar/src/Avatar.tsx:130]: Yeah, I don't love this. Imma rethink this logic a bit, it feels too much like I'm shoe-horning this in

Comment (serifluous) [inline packages/paste-core/components/avatar/src/Avatar.tsx:130]: question: why transparent and not color-background-body?

Comment (SiTaggart) [reply packages/paste-core/components/avatar/src/Avatar.tsx]: OK, this now feels much better

**Review by serifluous — APPROVED** [review]

**Review by TheSisb — APPROVED** [review]

**Pull request merged**